sleap/gui/commands.py

```diff
@@ -956,25 +953,53 @@ class ExportAnalysisFile(AppCommand):
     def do_action(cls, context: CommandContext, params: dict):
         from sleap.io.format.sleap_analysis import SleapAnalysisAdaptor
 
-        SleapAnalysisAdaptor.write(params["output_path"], context.labels)
+        for output_path, video in params["analysis_videos"]:
+            SleapAnalysisAdaptor.write(output_path, context.labels, video=video)
 
     @staticmethod
     def ask(context: CommandContext, params: dict) -> bool:
+        labels = context.labels
+        if len(labels.labeled_frames) == 0:
+            return False
+
+        if params["all_videos"]:
+            all_videos = context.labels.videos
+        else:
+            all_videos = [context.state["video"] or context.labels.videos[0]]
+
+        # Check for labeled frames in each video
+        videos = []
+        for video in all_videos:
+            lfs = labels.get(video)
+            if len(lfs) != 0:
+                videos.append(video)
```

Comment:
simplify with `videos = [video for video in all_videos if len(labels.get(video)) != 0]`